src/main/java/baritone/command/defaults/LitematicaCommand.java

```diff
@@ -40,11 +40,13 @@ public void execute(String label, IArgConsumer args) throws CommandException {
             if (args.is(Integer.class)) {
                 schematic = args.getAs(Integer.class) - 1;
             }
-        }
-        try {
-            baritone.getBuilderProcess().buildOpenLitematic(schematic);
-        } catch (IndexOutOfBoundsException e) {
-            logDirect("Pleas provide a valid index.");
+            try {
+                baritone.getBuilderProcess().buildOpenLitematic(schematic);
+            } catch (IndexOutOfBoundsException e) {
+                logDirect("Pleas provide a valid index.");
+            }
+        } else {
+            baritone.getBuilderProcess().buildOpenLitematic();
```

Comment:
`baritone.getBuilderProcess().buildOpenLitematic(LitematicaHelper.getSelectedIndex());`
this would remove the need of a second buildOpenLitematic method in BuilderProcess however it also wouldnt be wrong to remove the manual selection entirely and just keep the one that builds the schematic litematica has selected.

Reply:
jeah that's right, i added the second one with the idea that you can easily use it through the api, but of course you can just send the command directly to the execute functions, what works better anyways often.

Reply:
What about building placements by name? It's intuitive and with proper tab completion is't easy to use.

For easier API usage I think the best option would be a method taking the desired `SchematicPlacement` (Why are we even using indices in the first place?). Adding `buildOpenLitematic()` is good for consistency with `buildOpenSchematic()` though.

Reply:
names wont work well because you can have multiple schematics with the same name open. how would you decide which schematic is meant if you have 2 "myHouse.litematic"
as to why integers are used instead of SchematicPlacements thats because SchematicPlacement is a class from litematica and i didnt want to add a dependency to a optional mod to the BuilderProcess this way the only Class with a direct dependency on litematica should be LitematicaHelper and methods of that class should only be called after the LitematicaHelper.isLitematicaPresent() method returned true

Reply:
i added an branch with only the selected placement [thojo0/baritone/only_selected_litematic](/thojo0/baritone/tree/only_selected_litematic)
i will add a pull request if requested for that.

Reply:
> names wont work well because you can have multiple schematics with the same name open

We are talking about placements here and I though litematica enforces unique names on those, like it does for selection and subregion names. But apparently you can have indistinguishable placements. Reporting an error for duplicates would be an option, though it's certainly not as nice anymore.